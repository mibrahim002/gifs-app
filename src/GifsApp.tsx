import { CustomHeader } from './shared/components/CustomHeader'
import { SearchBar } from './shared/components/SearchBar'
import { PreviousSearches } from './gifs/components/PreviousSearches'
import { GifsList } from './gifs/components/GifsList'
import { useGifs } from './gifs/hooks/useGifs'

export const GifsApp = () => {
    const {previousTerms, gifs, handlerSearch, handleTermClicked} = useGifs()
  return (
    <>
        {/* Header */}
        <CustomHeader title='Buscador de Gifs'
        description='Descubre y comparte el Gifs perfecto'
        />

        {/* Search */}
       <SearchBar 
            placeholder='Busca lo que quieras'
            onQuery={handlerSearch}
            />

        {/* Previous Searches*/}
        <PreviousSearches 
            searches={previousTerms} 
            onLabelClick={handleTermClicked}
            />

        {/* Gifs */}
        <GifsList gifs={gifs}/>

    </>
  )
}
