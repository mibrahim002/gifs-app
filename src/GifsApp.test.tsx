import {describe, expect, test} from 'vitest'
import {render} from '@testing-library/react'
import { GifsApp } from './GifsApp'

describe('Gifs App', ()=> {
    test('shoul render component properly', ()=> {
        const {container} = render(<GifsApp/>)


        expect(container).toMatchSnapshot()
    })
})