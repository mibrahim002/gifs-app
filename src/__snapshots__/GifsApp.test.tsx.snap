// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Gifs App > shoul render component properly 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Buscador de Gifs
    </h1>
    <p>
      Descubre y comparte el Gifs perfecto
    </p>
  </div>
  <div
    class="search-container"
  >
    <input
      placeholder="Busca lo que quieras"
      type="text"
      value=""
    />
    <button>
      Buscar
    </button>
  </div>
  <div
    class="previous-searches"
  >
    <h2>
      Busquedada previas
    </h2>
    <ul
      class="previous-searches-list"
    />
  </div>
  <div
    class="gifs-container"
  />
</div>
`;
